data/core/common.lua

```diff
@@ -47,22 +47,26 @@ end
 
 
 function common.color(str)
-  local r, g, b, a = str:match("#(%x%x)(%x%x)(%x%x)")
+  local r, g, b, a = str:match("^#(%x%x)(%x%x)(%x%x)(%x?%x?)$")
```

Comment:
I don't like how this allows for single character alpha, but this is the limit of Lua patterns I guess.
It could be done the same way it's done for the `rgba...` part, or even using regex; but in the end it's still fine like this.

Reply:
> I don't like how this allows for single character alpha

It should be so. I've seen colors with one symbol transparency a couple of times.

> but this is the limit of Lua patterns I guess.

And that too. Lua has very limited regular expressions. I personally ran into a shortage several times: grouping characters into one class, n-times quantifier(where n>1), look back and forth. Although there are advantages like `%bxy`